internal/http/http.go

```diff
@@ -28,3 +32,41 @@ func SetCacheHeaders(w http.ResponseWriter, maxAge time.Duration, timeUntilExpir
 	w.Header().Set("Cache-Control", "public, max-age="+maxAge.String())
 	w.Header().Set("Expires", time.Now().Add(timeUntilExpiry).Format(time.RFC1123))
 }
+
+func LimitRead(r io.Reader) ([]byte, error) {
+	lr := io.LimitReader(r, maxBytes)
+
+	// Read as far as we are allowed to
+	bodyBytes, err := io.ReadAll(lr)
+	if err != nil {
+		return nil, Error(
+			fmt.Errorf(
+				"failed to read from reader: %w", err,
+			),
+			http.StatusBadRequest,
+		)
+	}
+
+	// If we read exactly the maximum, the body might be larger
+	if len(bodyBytes) == maxBytes {
+		// Try to read one more byte
+		buf := make([]byte, 1)
+		var n int
+		if n, err = r.Read(buf); err != nil && err != io.EOF {
+			return nil, Error(
+				fmt.Errorf(
+					"failed to check for additional content: %w",
+					err,
+				),
+				http.StatusInternalServerError,
+			)
```

Comment:
Don't need to wrap this since we know we treat all unclassified errors as internal server errors.

```suggestion
			return nil, fmt.Errorf("failed to check for additional content: %w", err)
```